### DISCUSSION TITLE
Is there a cheat sheet for the biome.json config?

### DISCUSSION THREAD

**Discussion by michealroberts** [biomejs/biome#3568] [Q&A]
Hi there 👋 

I've tried to use the codemod to transform my current prettier and eslint rules to biome.

This process did not go very well at all. Poorly formatted json files. Rules didn't transfer.

I'm not trying to reference the rules to allow formatting for JS and JSON, but I can't seem to get them working as before.

These were my old configuration files:

```yaml
bracketSpacing: true
printWidth: 100
semi: false
trailingComma: 'none'
singleQuote: true
tabWidth: 2
useTabs: false
arrowParens: 'avoid'
```

and

```
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint", "prettier"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier"
  ],
  "rules": {
    "no-console": 1,
    "prettier/prettier": 2
  }
}
```

This is my new biome.json file (I am leaving it as it is formated, as biome won't auto-matically format it even though I have it enabled?):

```json
{
	"$schema": "https://biomejs.dev/schemas/1.8.3/schema.json",
	"formatter": {
    "enabled": true,
    "ignore": ["dist", "node_modules"]
  },
  "linter": {
    "enabled": true,
		"ignore": ["dist", "node_modules"]
	},
  "organizeImports": {
		"enabled": true
	},
	"overrides": [
		{
			"include": ["*.json"],
			"formatter": {
				"indentWidth": 2
			}
		}
	],
  "javascript": {
		"formatter": {
			"enabled": true,
			"lineWidth": 100,
			"semicolons": "asNeeded",
			"trailingComma": "none",
			"indentWidth": 2,
			"indentStyle": "tab",
			"arrowParentheses": "asNeeded",
			"quoteStyle": "single"
		}
	},
  "json": {
    "formatter": {
      "enabled": true,
      "indentWidth": 2
    },
    "linter": {
      "enabled": true
    }
  }
}
```

**Comment by Conaclos** [comment]
Have you tried the migration tools that Biome embed? We have a [dedicated page about them](https://biomejs.dev/guides/migrate-eslint-prettier/).

We also have a [reference (kind of cheat sheet) for the configuration file](https://biomejs.dev/reference/configuration/).

**Reply by michealroberts** [reply]
@Conaclos Appreciate the quick response.

Yes, the migration tools crash quite often and therefore do not work as documented.

Running `npx @biomejs/biome migrate prettier --write`

```
Error: prettier is not expected in this context
```

Running `Error: eslint is not expected in this context`:

```
Error: eslint is not expected in this context
```

This has not been the best of DX's and introduction to biome, I'm not going to lie. But happy to work with you to resolve all of this.

I have added a biome.json file to my root and manually recreated my rules, and yet biome.json is left unformatted also.

**Reply by Conaclos** [reply]
Are you able to provide a reproduction? It could be great to open a bug report with the detailed info. I don't remember seeing this error.

> I have added a biome.json file to my root and manually recreated my rules, and yet biome.json is left unformatted also.

`biome format biome.json` should format the file.

**Reply by ematipico** [reply]
@michealroberts I noticed that your configuration file (old) is in YAML.

Biome doesn't support YAML format for prettier, only JSON.

**Reply by michealroberts** [reply]
@ematipico Hmmm, when running format or lint ... even though I feel I have the same rules applied, biome is creating havoc ... this is not a compatible API with prettier or eslint imho as described on the biome site, there are so many differences in the CLI, the rulesets, the ruleset names, the structure of the biome.json file etc

What would be the recommendations for this `eslint.json`:

```
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint", "prettier"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier"
  ],
  "rules": {
    "no-console": 1,
    "prettier/prettier": 2
  }
}
```

And this prettierrc.yaml file:

```
bracketSpacing: true
printWidth: 100
semi: false
trailingComma: 'none'
singleQuote: true
tabWidth: 2
useTabs: false
arrowParens: 'avoid'
```

**Reply by michealroberts** [reply]
Also, how do we navigate this:

```
The number of diagnostics exceeds the number allowed by Biome.
```

?

**Reply by ematipico** [reply]
@michealroberts 

I am not sure how much you're familiar with Biome, where you're coming from, and if you're read the documentation.

Biome formatter is pretty much compatible with Prettier.
Biome linter *isn't a drop-in replacement* of ESLint. We have different recommended rules. Our migrator is a **best effort** tool, so things might not be the same. I suggest your to get familiar with this resource, where we map ESLint rules to our rules: https://biomejs.dev/linter/rules-sources/ Inspired rules might work differently from the original.

> The number of diagnostics exceeds the number allowed by Biome.

You can use `--max-diagnostics`. You can navigate options by using `--help` or the reference page: https://biomejs.dev/reference/cli/

**Reply by michealroberts** [reply]
@ematipico Some compatibility differences I can list off the top of my head:

`arrowParens` becomes `arrowParenthesis`
`trailingComma` becomes `trailingCommas`
`semi` becomes `semicolons`

CLI --apply becomes --write ... etc

I guess by 97% compatibility is that for roughly every prettier rule there is an equivalent biome rule ... 

I read through the documentation before coming here, set everything up ... and, boop, .json is not formatting at all, be that biome.json or package.json. I think you definitely have some issues that haven't filtered up ...

**Comment by reecube** [comment]
The migration tool did not work for me. But I've created a full configuration file based on the docs and then went through all the configs to adapt them to my needs.

Full configuration file: https://gist.github.com/reecube/cae50c2daf2306399425f2516a79862b

Docs: https://biomejs.dev/reference/configuration/

**Reply by Conaclos** [reply]
> The migration tool did not work for me

Which issues have you met?
